bibliopixel/animation/animation.py

```diff
@@ -66,6 +67,8 @@ def cleanup(self, clean_layout=True):
     def compute_state(self):
         if self.threading.stop_event.isSet():
             self.state = STATE.canceled
+        elif self.runner.seconds and (time.time() - self.runner.run_start_time) > self.runner.seconds:
```

Comment:
Previous handling of `seconds` was all wrong. It depended on `fps` being specified which is not required if you want the animation to run unbounded or for some special animations like `ImageAnim` which get the delay value from the GIF file itself and act accordingly.  It also did not account for frames taking longer then the specified `fps` value. This new method respects time above all else.